portico: Add a loading indicator and rename Sign in to Log in. #6477 #7884

VishalCR7 opened the pull request:
Renamed Sign in to Login as mentioned in issue #7884 
Added a loading indicator when user clicks on Log in button.
Fixes #6477 #7884 .
Here's how it looks after clicking submit button:
![image](https://user-images.githubusercontent.com/24241258/34334048-dcb5cce2-e968-11e7-85af-ba3230f12ddc.png)

tiaozhiyaoyao commented on templates/zerver/login.html:
```diff
@@ -113,7 +113,7 @@ <h1 class="get-started">{{ _("Sign in to Zulip") }}</h1>
                     </div>
                     {% endif %}
 
-                    <button type="submit" name="button" class="full-width">{{ _("Sign in") }}</button>
```
try

VishalCR7 commented:
@rishig  Does it look good?  Any changes needed?

Also just wanted to know how a PR is merged in zulip after review.

rishig commented:
Thanks Vishal! You can check the "Assignees" on the right hand side to see who is assigned to review. It can take a few days sometimes, depending on when the reviewer is available.

The reviewer who merges it will leave a note here when it is merged.

brockwhittaker commented:
Some other feedback:

1. Do not put a background color on the loader. The button background changes colors so the loader looks incorrect on hover.

2. You should probably have the image always be there with visibility hidden or something because the only time you see this is on a slow connection, and on a slow connection this won't load. :) Maybe do this with an SVG instead and embed it, and display/hide when needed/not needed.

VishalCR7 commented:
@brockwhittaker as suggested, I have replaced the gif with a SVG and embedded it. Please review it let me know if it requires some more changes.

VishalCR7 commented:
@rishig  can you please explain why codecov/project check is failing? I can't figure it out from the error details.

rishig commented:
I suspect the codecov is complaining because of the large svg file being added. Feel free to ignore that error. 
I'll let Brock continue with the rest of the review. Thanks @VishalCR7!

akashnimare commented:
Suggestion - How about removing "Log in" when user clicks on "Log in" button and just show the loading indicator - 

<img src="https://d2ffutrenqvap3.cloudfront.net/items/3A2l1k2p1E2o2D0D3O0q/Screen%20Recording%202017-12-29%20at%2008.32%20PM.gif?v=a04808a5"/>

rishig commented:
Agreed with Akash, we can remove the "Logging in .." words when this is merged. We should probably also keep it as "Sign in" until the Google/GitHub buttons are changed. Other than that is this ready to merge @brockwhittaker?

VishalCR7 commented:
@rishig @brockwhittaker  I have renamed login again to sign in and removed the loging in text. hope it looks good now. Please review it let me know about any further changes if any. Thanks

timabbott commented:
The loading indicator was done in #6477, and I think #7884 tracks the sign in work.  So closing as obsolete, thanks @VishalCR7.
